src/bokeh/io/export.py

```diff
@@ -524,49 +521,15 @@ def _SVG_SCRIPT(obj: Model | Document) -> str:
   doc.idle.connect(done);
 """
 
-
-class _TempFile:
-    _closed: bool = False
-
-    fd: int
-    path: str
-
-    def __init__(self, *, prefix: str = "tmp", suffix: str = "") -> None:
-        # XXX: selenium has issues with /tmp directory (or equivalent), so try using the
-        # current directory first, if writable, and otherwise fall back to the system
-        # default tmp directory.
-        try:
-            self.fd, self.path = mkstemp(prefix=prefix, suffix=suffix, dir=os.getcwd())
-        except OSError:
-            self.fd, self.path = mkstemp(prefix=prefix, suffix=suffix)
-
-    def __enter__(self) -> _TempFile:
-        return self
-
-    def __exit__(self, exc: Any, value: Any, tb: Any) -> None:
-        self.close()
-
-    def __del__(self) -> None:
-        self.close()
-
-    def close(self) -> None:
-        if self._closed:
-            return
-
-        try:
-            os.close(self.fd)
-        except OSError:
-            pass
-
-        try:
-            os.unlink(self.path)
-        except OSError:
-            pass
-
-        self._closed = True
-
-def _tmp_html() -> _TempFile:
-    return _TempFile(prefix="bokeh", suffix=".html")
+@contextmanager
+def _tmp_html() -> Iterator[_TemporaryFileWrapper[bytes]]:
+    # according to https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile
+    # in order for named temp files to be safely re-openable on Windows, we need
+    # to set delete=False, so explicitly this context manager is for explicitly
+    # managing the unlink after we are done.
+    tmp = NamedTemporaryFile(mode="wb", prefix="bokeh", suffix=".html", delete=False)
+    yield tmp
+    os.unlink(tmp.name)
```

Comment:
```suggestion
    try:
        yield tmp
    finally:
        os.unlink(tmp.name)
```

So if something errors out we don't keep the tmp file. Irrelevant for Linux, not sure how often it is cleaned up on Windows/Mac.

Reply:
Right I was recalling how `__exit__` always runs but forgot `@contextmanager` is different [37e66e8](https://github.com/bokeh/bokeh/pull/14640/commits/37e66e8ceab404b8e38b1cd6eac9cacb4022ed9c)